src/libraries/Microsoft.PowerFx.Core/Lexer/Tokens/IdentToken.cs

```diff
@@ -70,7 +72,28 @@ internal override Token Clone(Span ts)
         // REVIEW ragru: having a property for every possible error isn't scalable.
         internal bool HasDelimiters => HasDelimiterStart;
 
-        internal bool HasErrors => IsModified || (HasDelimiterStart && !HasDelimiterEnd);
+        /// <summary>
+        /// Whether an identifier has errors.
+        /// </summary>
+        public bool HasErrors => IsModified || (HasDelimiterStart && !HasDelimiterEnd);
+
+        /// <summary>
+        /// Converts a string value of an identifier to a valid identifier (e.g., "a b" -> "'a b'").
+        /// </summary>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException">Throw if value is <c>null</c>.</exception>
+        public static string MakeValidIdentifier(string value)
```

Comment:
>MakeValidIdentifier [](http://example.com/codeflow?start=29&length=19)

It seems like this should have some failure path. 
 Ie, if we pass bad chars ("***") 

Related to: 
public static DName MakeValid(string strName, out bool fModified)

Reply:
It seems that the lexer is accepting anything really (e.g., in the web demo `'***'` can be used as a legal identifier). The only failure path is `null`. Or should we test this in some other way?